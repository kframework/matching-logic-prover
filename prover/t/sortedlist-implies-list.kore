imports "t/definitions/listSorted.kore"
imports "t/definitions/list.kore"


claim \implies( \and( listSorted( H { ArrayIntInt }
                                , X { Int }
                                , K { SetInt }
                                , MIN { Int }
                                )
                    )
              , \and( list( H { ArrayIntInt }
                          , X { Int }
                          , K { SetInt }
                          )
                    )
              )
strategy search-fol(bound: 3)
/*
strategy kt . ( (right-unfold-Nth(0, 0) . simplify . direct-proof)
              | ( simplify . direct-proof )
              | (right-unfold-Nth(0, 1) . simplify . direct-proof)
              )
*/
