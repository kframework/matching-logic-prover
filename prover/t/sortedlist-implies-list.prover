claim \implies( \and( listSorted( variable("H") { ArrayIntInt }
                                , variable("X") { Int }
                                , variable("K") { SetInt }
                                , variable("MIN") { Int }
                                , .Patterns
                                )
                    , .Patterns
                    )
              , \and( list( variable("H") { ArrayIntInt }
                          , variable("X") { Int }
                          , variable("K") { SetInt }
                          , .Patterns
                          )
                    , .Patterns
                    )
              )

strategy search-bound(5)

/*
strategy kt ; ( (right-unfold-Nth(0, 0) ; simplify ; direct-proof)
              | ( simplify ; direct-proof )
              | (right-unfold-Nth(0, 1) ; simplify ; direct-proof)
              )
*/
