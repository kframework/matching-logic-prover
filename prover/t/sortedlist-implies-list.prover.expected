<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( listSorted ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "K" ) { SetInt } , variable ( "MIN" ) { Int } , .Patterns ) , .Patterns ) , \and ( list ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "K" ) { SetInt } , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
